common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala

```diff
@@ -3457,4 +3466,78 @@ object CelebornConf extends Logging {
       .doc("enable the partition split in worker side")
       .booleanConf
       .createWithDefault(true)
+
+  // Flink specific client configurations.
+  val MEMORY_PER_PARTITION_MIN: ConfigEntry[Long] =
+    buildConf("celeborn.client.partition.minMemory")
+      .withAlternative("remote-shuffle.job.min.memory-per-partition")
+      .categories("client")
+      .version("0.3.0")
+      .doc("Min memory reserved for a partition. For Flink plugin only.")
+      .bytesConf(ByteUnit.BYTE)
+      .createWithDefault(8 * 1024 * 1024)
+
+  val MEMORY_PER_INPUT_GATE_MIN: ConfigEntry[Long] =
+    buildConf("celeborn.client.inputGate.minMemory")
+      .withAlternative("remote-shuffle.job.min.memory-per-gate")
+      .categories("client")
+      .doc("Min memory reserved for a InputGate. For Flink plugin only.")
+      .version("0.3.0")
+      .bytesConf(ByteUnit.BYTE)
+      .createWithDefault(8 * 1024 * 1024)
+
+  val NUM_CONCURRENT_READINGS: ConfigEntry[Int] =
+    buildConf("celeborn.client.flink.inputGate.concurrentReadings")
+      .withAlternative("remote-shuffle.job.concurrent-readings-per-gate")
+      .categories("client")
+      .version("0.3.0")
+      .doc("Max concurrent reading channels for a input gate. For Flink plugin only.")
+      .intConf
+      .createWithDefault(Int.MaxValue)
+
+  val MEMORY_PER_PARTITION: ConfigEntry[Long] =
+    buildConf("celeborn.client.flink.partition.memory")
+      .withAlternative("remote-shuffle.job.memory-per-partition")
+      .categories("client")
+      .version("0.3.0")
+      .doc("Memory reserved for a partition. For Flink plugin only.")
+      .bytesConf(ByteUnit.BYTE)
+      .createWithDefault(64 * 1024 * 1024)
+
+  val MEMORY_PER_INPUT_GATE: ConfigEntry[Long] =
+    buildConf("celeborn.client.flink.inputGate.memory")
+      .withAlternative("remote-shuffle.job.memory-per-gate")
+      .categories("client")
+      .version("0.3.0")
+      .doc("Memory reserved for a input gate. For Flink plugin only.")
+      .bytesConf(ByteUnit.BYTE)
+      .createWithDefault(32 * 1024 * 1024)
+
+  val INPUT_GATE_SUPPORT_FLOATING_BUFFER: ConfigEntry[Boolean] =
+    buildConf("celeborn.client.inputGate.supportFloatingBuffer")
+      .withAlternative("remote-shuffle.job.support-floating-buffer-per-input-gate")
+      .categories("client")
+      .version("0.3.0")
+      .doc("Whether to support floating buffer in Flink input gates.")
+      .booleanConf
+      .createWithDefault(true)
+
+  val DATA_COMPRESSION_ENABLED: ConfigEntry[Boolean] =
+    buildConf("celeborn.client.compression.enabled")
```

Comment:
why not reuse `celeborn.client.shuffle.compression.codec`?

Reply:
`celeborn.client.shuffle.compression.codec` should allow "none" as value

Reply:
This config will be reserved for compatibility. `celeborn.client.compression.enabled` is already in use now.

Reply:
it's for 0.3.0, what does you mean "already"?

Reply:
One supporting Flink's reason is to replace an online shuffle service called flink-remote-shuffle.  These configs are already used by workload.

Reply:
OK, I see